packages/util/src/signature.ts

```diff
@@ -46,6 +46,9 @@ export function ecsign(msgHash: Buffer, privateKey: Buffer, chainId: any): any {
 
 function calculateSigRecovery(v: BNLike, chainId?: BNLike): BN {
   const vBN = toType(v, TypeOutput.BN)
+
+  if (vBN.eqn(0) || vBN.eqn(1)) return toType(v, TypeOutput.BN)
```

Comment:
Ok, this is the central line of this PR (this is just an additional transparency explanation for everyone to follow including the team). So this allows `v` to directly pass through if `0` or `1`. The `chainId` parameter is ignored in this case.

The function here is not exported. Effects are indirect, to all methods below this comment where additional `NOTE:` line comments are added. I've checked all the methods and do not see security or other implications or side effects.

Namely:

- `ecrecover`: This will then allow with this update to directly pass in an EIP-1559 (or any other typed) tx with v being `0` or `1` and will recover the correct PK in these cases as well. No side effects on other (legacy) txs.
- `toRpcSig`: Now allows EIP-1559 txs to be converted in `eth_sign` format, no side effects
- `toCompactSig`: Now allows EIP-1559 txs to be converted in `eth_sign` format, no side effects (was first a bit unsure about the `0` case but this is included as well respectively not affected)
- `isValidSignature`: Now allows EIP-1559 txs to be converted in `eth_sign` format, no side effects

So I would give this a go and merge this and would include in the very imminent last round of `master` releases before we merge in `develop`.

We can then continue on `develop` with do some further signature code cleanup - with some inspiration from the breaking-release-TODOs - in favor of a complete direct removal.